syncing/src/main/kotlin/maru/syncing/ELSyncService.kt

```diff
@@ -85,21 +92,37 @@ class ELSyncService(
         blockNumber = latestBeaconBlockBody.executionPayload.blockNumber,
         blockHash = latestBeaconBlockBody.executionPayload.blockHash,
       )
-    log.debug("New EL sync target = {}", newELSyncTarget)
+    log.debug("New elSyncTarget={}", newELSyncTarget)
 
     val finalizationState = finalizationProvider(latestBeaconBlockBody)
     val fcuResponse =
       executionLayerManager
-        .setHead(
-          headHash = newELSyncTarget.blockHash,
-          safeHash = finalizationState.safeBlockHash,
-          finalizedHash = finalizationState.finalizedBlockHash,
-        ).get()
+        .newPayload(latestBeaconBlockBody.executionPayload)
```

Comment:
Is this `newPayload` call a must to sync the EL? If not, why add here? Its a bit counter-intuitive to me, because how could EL validate the `latestBeaconBlockBody.executionPayload` (i.e. EL block 100) beforehand if it was not synced to that point (i.e. still at EL block 80)? Or I'm misunderstanding sth?

Reply:
Oh I found [this](https://github.com/danielrachi1/diagrams/blob/main/rendered/engine-api/newPayloadV2.png), and seems like if the EL doesn't have the required block data to validate the new payload, it would automatically switch to SYNCING mode (for some clients)